--- pycurl-7.19.3.1/setup.py.orig	2014-02-08 22:45:35.815908728 +0100
+++ pycurl-7.19.3.1/setup.py	2014-02-09 07:30:52.621252762 +0100
@@ -147,8 +147,8 @@
         optbuf = ''
         sslhintbuf = ''
         errtext = ''
-        for option in ["--libs", "--static-libs"]:
-            p = subprocess.Popen((CURL_CONFIG, option),
+        for option in ["--libs", "--static --libs"]:
+            p = subprocess.Popen("'%s' %s libcurl" % ('pkg-config', option), shell=True,
                 stdout=subprocess.PIPE, stderr=subprocess.PIPE)
             stdout, stderr = p.communicate()
             if p.wait() == 0:
